### PR TITLE
Close config file object (deferred close)

### PR DESCRIPTION
**Pull request by lyubenblagoev**
Config file objects will be closed by the GC, but using the defer statement is always preferable.

### CODE REVIEW

**Comment by rdowner** [comment]
Hi @lyubenblagoev and welcome to our project :smile:

The Apache Software Foundation prefers contributors to have submitted a Contributor License Agreement, but for this small change it's not necessary IMO. If you would like to contribute more in the future, please consider submitting an ICLA, [as described here](https://www.apache.org/licenses/#clas).

Our Go experts @geomacy @robertgmoss, do you have any comments on this PR?

File: io/config.go [hunk 1]
```diff
@@ -20,9 +20,10 @@ package io
 
 import (
 	"encoding/json"
-	"github.com/apache/brooklyn-client/error_handler"
 	"os"
 	"path/filepath"
+
+	"github.com/apache/brooklyn-client/error_handler"
```
Comment (rdowner) [inline io/config.go]: Is this change intentional?

Comment (lyubenblagoev) [inline io/config.go]: This came from the `gofmt` tool used by the vim-go plugin. It organizes imports automatically by path and third party packages are separated with blank line from standard library packages.

File: io/config.go [hunk 2]
```diff
@@ -48,12 +49,12 @@ func GetConfig() (config *Config) {
 }
 
 func (config *Config) Write() {
-
 	// Create file as read/write by user (but does not change perms of existing file)
 	fileToWrite, err := os.OpenFile(config.FilePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
 	if err != nil {
 		error_handler.ErrorExit(err)
 	}
+	defer fileToWrite.Close()
```
Comment (geomacy) [inline io/config.go]: Should this be inside the "if" statement above?   What does OpenFile return in the first arg if the second is an err?

File: io/config.go [hunk 3]
```diff
@@ -64,6 +65,8 @@ func (config *Config) Read() {
 	if err != nil {
 		error_handler.ErrorExit(err)
 	}
+	defer fileToRead.Close()
```
Comment (geomacy) [inline io/config.go]: same here of course

File: io/config.go (hunk 2 above)
Comment (lyubenblagoev) [inline io/config.go]: Nope. `os.OpenFile` will return the file object and nil as an error, or nil and the error in case of an error.

File: io/config.go (hunk 3 above)
Comment (lyubenblagoev) [inline io/config.go]: The same as above file; `os.OpenFile` returns file, nil or nil, err, but not both file object and error.

File: io/config.go (hunk 2 above)
Comment (geomacy) [inline io/config.go]: ah I wasn't reading closely - of course the "if" handles the  !=  case!

**Comment by geomacy** [comment]
+1 looks good to me,  many thanks for the contribution @lyubenblagoev !

**Comment by robertgmoss** [comment]
LGTM

**Comment by aledsage** [comment]
Thanks @lyubenblagoev - are you happy to submit a standard Contributor License Agreement, as described by @rdowner above? Can you let us know when you've done that and we'll very happily merge this!

**Comment by lyubenblagoev** [comment]
@aledsage, I've completed and submitted ICLA by email, as described in the link provided by @rdowner.
